src/hit/api/params.cpp

```diff
@@ -3,10 +3,105 @@
 
 #include "params.h"
 
+#include <boost/multiprecision/miller_rabin.hpp>
+
 using namespace std;
 using namespace seal;
+using namespace boost::multiprecision;
 
 namespace hit {
+
+    // This is _not_ adversarial prime generation. The primes will be public, and security is based on
+    // the total bit size of the _product_ of each "prime", so there are no security implications if
+    // we accidentally generate a composite.
+    int miller_rabin_iters = 25;
+
+    // true if x \in mods, false otherwise
+    bool contains(uint64_t x, const vector<Modulus> &mods) {
+        for (const auto &m : mods) {
+            if (x == m.value()) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
Should we use build-in methods?
```suggestion
        return std::find(mods.begin(), mods.end(), x) != mods.end();
```